addons/website_slides/controllers/main.py

```diff
@@ -368,3 +377,104 @@ def slides_embed(self, slide_id, page="1", **kw):
         except AccessError: # TODO : please, make it clean one day, or find another secure way to detect
                             # if the slide can be embedded, and properly display the error message.
             return request.render('website_slides.embed_slide_forbidden', {})
+
+    # Profile
+    # ---------------------------------------------------
+    def _prepare_open_slide_user(self, user):
+        slides = request.env['slide.slide'].sudo().search([('user_ids', '=', user.id)])
+        courses = request.env['slide.channel.partner'].sudo().search([('partner_id', '=', user.partner_id.id)]).channel_id
+        courses_completed = courses.filtered(lambda c: c.completed)
+        courses_ongoing = courses - courses_completed
+
+        values = {
+            'uid': request.env.user.id,
+            'user': user,
+            'main_object': user,
+            'slides': slides,
+            'count_slides': len(slides),
+            'courses': courses,
+            'count_courses': len(courses),
+            'is_profile_page': True,
+        }
+        return values
+
+    @http.route(['/slides/user/<int:user_id>'], type='http', auth="public", website=True)
+    def open_cross_slide_user(self, user_id=0, **post):
+        user = request.env['res.users'].sudo().browse([user_id])
+
+        if post.get('channel'):
+            channels = post.get('channel')
+        elif post.get('channel_id'):
+            channels = request.env['slide.channel'].browse(int(post.get('channel_id')))
+        else:
+            channels = request.env['slide.channel'].search([])
+
+        values = {
+            'user': request.env.user,
+            'is_public_user': request.env.user.id == request.website.user_id.id,
+            'notifications': self._get_notifications(),
+            'header': post.get('header', dict()),
+            'searches': post.get('searches', dict()),
+            'validation_email_sent': request.session.get('validation_email_sent', False),
+            'validation_email_done': request.session.get('validation_email_done', False),
+            'channel': channels[0] if len(channels) == 1 else True,
+        }
+        values.update(self._prepare_open_slide_user(user))
+
+        return request.render("website_slides.user_detail_cross_full", values)
+
+    @http.route(['/slides/<model("slide.channel"):channel>/user/<int:user_id>'], type='http', auth="public", website=True)
+    def open_slide_user(self, channel, user_id=0, **post):
+        user = request.env['res.users'].sudo().browse(user_id)
+        current_user = request.env.user.sudo()
+
+        # Users with high karma can see users with karma <= 0 for
+        # moderation purposes, IFF they have posted something (see below)
+        if (not user or (user.karma < 1 and current_user.karma < channel.karma_unlink_all)):
```

Comment:
Post-commented thingy: this is an example of why we should have a unique check method to access a user profile so that we don't have custom implementation in various controllers :) .